sdk/tables/data-tables/src/utils/connectionString.ts

```diff
@@ -95,10 +102,14 @@ function getValueInConnString(
     | "EndpointSuffix"
     | "SharedAccessSignature"
 ): string {
+  const searchKey = argument.toLowerCase();
   const elements = connectionString.split(";");
   for (const element of elements) {
-    if (element.trim().startsWith(argument)) {
-      return element.trim().match(argument + "=(.*)")![1];
+    const trimmedElement = element.trim();
+    const [elementKey, value] = trimmedElement.split("=");
+    const key = elementKey.toLowerCase();
+    if (key.startsWith(searchKey)) {
```

Comment:
I think the semantics here changed slightly, since in the past we would have required an exact match whereas now we do a prefix match?

In practice I don't think we can run into an issue today because none of the arguments are prefixes of one another, but could we just do `if (key === searchKey)` instead?  I think it would be more clear?

Reply:
Yeah that's a good call. I'll update this. Thanks for pointing it out @ellismg